subprojects/docs/src/snippets/configurationCache/testKit/groovy/src/test/groovy/org/example/BuildLogicFunctionalTest.groovy

```diff
@@ -0,0 +1,50 @@
+package org.example
+
+import org.gradle.testkit.runner.GradleRunner
+import org.junit.Rule
+import org.junit.rules.TemporaryFolder
+import spock.lang.Specification
+
+import static org.gradle.testkit.runner.TaskOutcome.*
+
+class BuildLogicFunctionalTest extends Specification {
+
+    @Rule TemporaryFolder testProjectDir = new TemporaryFolder()
+    File buildFile
+
+    def setup() {
+        testProjectDir.newFile('settings.gradle') << ""
+        buildFile = testProjectDir.newFile('build.gradle')
+    }
+
+    // tag::functional-test-configuration-cache[]
+    def "my task can be loaded from the configuration cache"() {
+        given:
+        buildFile << """
+            plugins {
+                id 'org.example.my-plugin'
+            }
+        """
+
+        when:
+        runner()
+            .withArguments( '--configuration-cache=on', 'myTask')   // <1>
```

Comment:
⭕ If `--configuration-cache` is enough, I'd rather use it.

Reply:
We don't yet, see #13170